storage/src/main/java/com/brandwatch/ahmedanalytics/storage/StorageJob.java

```diff
@@ -0,0 +1,60 @@
+package com.brandwatch.ahmedanalytics.storage;
+
+import com.brandwatch.ahmedanalytics.common.entities.Mention;
+import com.brandwatch.ahmedanalytics.common.repositories.MentionsRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class StorageJob extends Thread{
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final KafkaConsumer<String, String> mentionConsumer;
+
+    @Autowired
+    private MentionsRepository mentionsRepository;
+
+    public StorageJob(KafkaConsumer<String, String> mentionConsumer) {
+        this.mentionConsumer = mentionConsumer;
+        this.mentionConsumer.subscribe(Collections.singletonList("mention"));
```

Comment:
`Subscribe` should be called inside `run()` method to be sure all the calls to mention consumer are done from the same thread because `KafkaConsumer` is not thread safe.

Reply:
Can we also make topic name injected via constructor parameter? It's almost always better to have configurable names.

Reply:
I actually did not get everything about the thread thing. I do not know how a race condition will occur and how we solved it by doing that. anyway I made the subscription in the run() method.
for the topic name, do u mean that I should make it a bean?